### ISSUE TITLE
Modifying array element value breaks reactivity

### ISSUE THREAD

**Issue by windymindy** [monterail/vuelidate#404]
https://jsfiddle.net/windymindy/u0kq9Lrb/

Subsequent '$each.$model' changes are not reactive:
1. Input into a text field.
2. Make sure '<pre>{{ $v }}</pre>' contains your new input.
3. Modify input again. From now on updates won't trigger changes in displayed text.
4. Press 'check model' button to confirm, that data was, however, updated.

**Comment by venticco** [comment]
What breaks reactivity is direct usage of array's element with `v-model` boundary.

According to [documentation](https://vuejs.org/v2/guide/list.html#Caveats) changing an element directly with `array[index]` syntax wont result in reactivity update (`v-model` in your fiddle uses `"$v.array.$each[0].$model"`). 

In order to work properly your current approach with `v-model` should be changed to `:value` prop and `@input` event handler with `Vue.set` method mutating array. 

This is your fiddle modified with described changes: https://jsfiddle.net/kjmonterail/u0kq9Lrb/14/

**Comment by windymindy** [comment]
I insist, there is a bug. $v.array.$each[0] is an object. Modifying it's property $model should lead to value change and $dirty flag flip. It actualy happens in the reproducible, but only once.

**Comment by venticco** [comment]
Ok, now I get it. This actually may be a bug, but still needs more investigation about the problem

**Comment by jean-moldovan** [comment]
Interestingly enough it works ok if you use `v-model` in the array item directly (without $model):
https://jsfiddle.net/cyrn68m4/2/

**Comment by windymindy** [comment]
Of course, it works, because no vuelidate is used in this example. $v just renders changed data.
It has got to be direct array write instead of splice somewhere in vuelidate/src.index.js when assigning to model. I just can't find it yet.

**Comment by windymindy** [comment]
Okay, so here is the code.
https://github.com/monterail/vuelidate/blob/f5b33d0ebe0c0752fae47b1f6bcfa25055a6bde6/src/index.js#L386
It treats every type of data the same way: just sets the _prop_ of a _parent_ model using _[]_ operator. And in case type of _parent_ is _Array_ things break.

**Comment by windymindy** [comment]
The first example shows, that the library doesn't support collections of primitives (the docs contain object collection example). The scenario to reproduce is the same as in the opening comment.
https://jsfiddle.net/windymindy/pu1o3tb9/

The second example contains a quirk.
https://jsfiddle.net/windymindy/1qed7rhc/

Is there a cleaner way to patch this???

**Comment by venticco** [comment]
@windymindy thank you for all the work you've done on this! You can use [patch-package](https://www.npmjs.com/package/patch-package) in your project applying local patch until new version with fix is released.
